thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalySummaryGenerator.java

```diff
@@ -0,0 +1,169 @@
+
+import com.linkedin.thirdeye.api.dto.MergedAnomalyResult;
+import com.linkedin.thirdeye.db.entity.AnomalyResult;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public abstract class AnomalySummaryGenerator {
+
+  private AnomalySummaryGenerator() {
+
+  }
+
+  /**
+   * @param anomalies
+   * @param mergeConfig
+   *
+   * @return
+   */
+  public static List<MergedAnomalyResult> mergeAnomalies(List<AnomalyResult> anomalies,
+      AnomalyMergeConfig mergeConfig) {
+    return mergeAnomalies(null, anomalies, mergeConfig);
+  }
+
+  /**
+   * @param mergedAnomaly
+   * @param anomalies
+   * @param mergeConfig
+   *
+   * @return
+   */
+  public static List<MergedAnomalyResult> mergeAnomalies(MergedAnomalyResult mergedAnomaly,
+      List<AnomalyResult> anomalies, AnomalyMergeConfig mergeConfig) {
+
+    // sort anomalies in natural order of start time
+    Collections
+        .sort(anomalies, (o1, o2) -> (int) ((o1.getStartTimeUtc() - o2.getStartTimeUtc()) / 1000));
+
+    boolean applySequentialGapBasedSplit = false;
+    boolean applyMaxDurationBasedSplit = false;
+
+    if (mergeConfig.getMergeDuration() > 0) {
+      applyMaxDurationBasedSplit = true;
+    }
+
+    if (mergeConfig.getSequentialAllowedGap() > 0) {
+      applySequentialGapBasedSplit = true;
+    }
+
+    List<MergedAnomalyResult> mergedAnomalies = new ArrayList<>();
+
+    for (int i = 0; i < anomalies.size(); i++) {
+      AnomalyResult currentResult = anomalies.get(i);
+      if (mergedAnomaly == null) {
+        mergedAnomaly = new MergedAnomalyResult();
+        populateMergedResult(mergedAnomaly, currentResult);
+        mergedAnomaly.setMergeStrategy(mergeConfig.getMergeStrategy());
+      } else {
+        // check if Max Duration for merged has passed, if so, create new one
+        if (applyMaxDurationBasedSplit) {
+          if (mergedAnomaly.getEndTime() - mergedAnomaly.getStartTime() >= mergeConfig
+              .getMergeDuration()) {
+            // Split here
+            mergedAnomalies.add(mergedAnomaly);
+            mergedAnomaly = new MergedAnomalyResult();
+            populateMergedResult(mergedAnomaly, currentResult);
+            mergedAnomaly.setMergeStrategy(mergeConfig.getMergeStrategy());
+          }
+        }
+
+        // compare current with merged and decide whether to merge the current result or create a new one
+        if (applySequentialGapBasedSplit
+            && (currentResult.getStartTimeUtc() - mergedAnomaly.getEndTime()) > mergeConfig
+            .getSequentialAllowedGap()) {
+          // Split here if not processed already
+          mergedAnomalies.add(mergedAnomaly);
+          mergedAnomaly = new MergedAnomalyResult();
+          populateMergedResult(mergedAnomaly, currentResult);
+          mergedAnomaly.setMergeStrategy(mergeConfig.getMergeStrategy());
```

Comment:
isn't the merge strategy being taken into account anywhere? it looks like a pure time based merge, or am i missing something